articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+
+The `MatrixTransform` is computed by the [`SpriteEffect`](xref:Microsoft.Xna.Framework.Graphics.SpriteEffect) class. The full source is available, [`here`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Effect/SpriteEffect.cs#L63). The relevant lines are copied below:
+
+```csharp
+// cache the shader parameter for the MatrixTransform
+_matrixParam = Parameters["MatrixTransform"];
+
+// ... some code left out for readability
+
+// create a projection matrix in the _projection variable
+Matrix.CreateOrthographicOffCenter(0, vp.Width, vp.Height, 0, 0, -1, out _projection);
+
+// ... some code left out for readability 
+
+// assign the projection matrix to the MatrixTransform
+_matrixParam.SetValue(_projection);
+```
+
+There are two common types of project matrices, 
+1. Orthographic,
+2. Perspective
+
+The orthographic projection matrix produces the classic 2d sprite effect, where sprites have no perspective when they are on the sides of the screen. 
+
+
+## Custom Vertex Shader
+
+Now that you understand the default vertex shader being used by `SpriteBatch`, we can replace the shader with a custom shader. The new shader must accomplish the basic requirements, 
+1. convert the vertices from world-space to clip-space
+2. provide the input semantics required for the pixel shader.
+
+To experiment with this, create a new Sprite Effect called `3dEffect` in the _MonoGameLibrary_'s shared content effects folder. We need to add a vertex shader function. To do that, we need a new `struct` that holds all the input semantics passed from `SpriteBatch`: 
+
+> [!tip] 
+> Use a struct for inputs and outputs.
+> 
+> The default vertex shader accepts all 3 inputs (`position`, `color`, and `texCoord`) as direct parameters. However, when you have more than 1 semantic, it is helpful to organize all of the inputs in a `struct`. 
+
+```hlsl
+struct VertexShaderInput
+{
+    float4 Position	: POSITION0;
+    float4 Color	: COLOR0;
+    float2 TexCoord	: TEXCOORD0;
+};
+```
+
+Now add the stub for the vertex shader function:
+
+```hlsl
+VertexShaderOutput MainVS(VertexShaderInput input) 
+{
+    VertexShaderOutput output;
+    
+    return output;
+}
+```
+
+And finally modify the `technique` to _include_ the vertex shader function. Until now, the `MainVS()` function is just considered as any average function in your shader, and since it wasn't used from the `MainPS` pixel shader, it would be compiled out of the shader. When you specify the `MainVS()` function as the vertex shader function, you are overriding the default `SpriteBatch` vertex shader function:
+
+```hlsl
+technique SpriteDrawing
+{
+	pass P0
+	{
+		PixelShader = compile PS_SHADERMODEL MainPS();
+		VertexShader = compile VS_SHADERMODEL MainVS();
+	}
+};
```

Comment:
Include the rest of the shader block and use line highlighting as shown here:

https://github.com/cdhanna/docs.monogame.github.io/blob/2aef6d967b5ed7ecc66816177f1532059b37ed24/articles/tutorials/building_2d_games/11_input_management/index.md?plain=1#L369